sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala

```diff
@@ -789,55 +789,50 @@ case class ShowCreateTableCommand(table: TableIdentifier) extends RunnableComman
   )
 
   override def run(sparkSession: SparkSession): Seq[Row] = {
-    val catalog = sparkSession.sessionState.catalog
-    val tableMetadata = catalog.getTableMetadata(table)
-
-    // TODO: unify this after we unify the CREATE TABLE syntax for hive serde and data source table.
-    val stmt = if (DDLUtils.isDatasourceTable(tableMetadata)) {
-      showCreateDataSourceTable(tableMetadata)
-    } else {
-      showCreateHiveTable(tableMetadata)
-    }
+    val tableMeta = sparkSession.sessionState.catalog.getTableMetadata(table)
+    val tableName = tableMeta.identifier.quotedString
 
-    Seq(Row(stmt))
-  }
-
-  private def showCreateHiveTable(metadata: CatalogTable): String = {
-    def reportUnsupportedError(features: Seq[String]): Unit = {
+    if (tableMeta.unsupportedFeatures.nonEmpty) {
       throw new AnalysisException(
-        s"Failed to execute SHOW CREATE TABLE against table/view ${metadata.identifier}, " +
+        s"Failed to execute SHOW CREATE TABLE against table/view $tableName, " +
           "which is created by Hive and uses the following unsupported feature(s)\n" +
-          features.map(" - " + _).mkString("\n")
-      )
+          tableMeta.unsupportedFeatures.map(" - " + _).mkString("\n"))
     }
 
-    if (metadata.unsupportedFeatures.nonEmpty) {
-      reportUnsupportedError(metadata.unsupportedFeatures)
-    }
+    val stmt = if (tableMeta.tableType == VIEW) {
+      val builder = StringBuilder.newBuilder
+      builder ++= s"CREATE VIEW $tableName"
 
-    val builder = StringBuilder.newBuilder
+      if (tableMeta.schema.nonEmpty) {
+        builder ++= tableMeta.schema.map(_.name).mkString("(", ", ", ")")
+      }
 
-    val tableTypeString = metadata.tableType match {
-      case EXTERNAL => " EXTERNAL TABLE"
-      case VIEW => " VIEW"
-      case MANAGED => " TABLE"
+      builder ++= s" AS\n${tableMeta.viewText.get}"
+      builder.toString
+    } else if (DDLUtils.isHiveTable(tableMeta) && tableMeta.properties.nonEmpty) {
```

Comment:
this condition is true most of the time, because Hive metastore will generate some table properties. https://issues.apache.org/jira/browse/SPARK-19241 is tracking it